### PR TITLE
feat: update status color usage

### PR DESCRIPTION
**Pull request by reidzeibel**
related to
- https://github.com/AtB-AS/kundevendt/issues/15379
- https://github.com/AtB-AS/kundevendt/issues/18138
- https://github.com/AtB-AS/kundevendt/issues/18136

related PR on https://github.com/AtB-AS/design-system/ :
- https://github.com/AtB-AS/design-system/pull/171
- https://github.com/AtB-AS/design-system/pull/172

This PR will update the usage of all `StatusColor` to match the new system.
Color update (when it is ready) will also affect the "New"/"Ny" badge color in https://github.com/AtB-AS/kundevendt/issues/18132

Most color usage right now is mapped to `primary` and some to `secondary`, currently there are no differences between `primary` and `secondary`, but this will change when the design for OMS is ready, and the usage should be checked across components.

<details>
<summary>Changes on Status color type</summary>

#### From
```
export type ContrastColor = {
  background: string;
  text: string;
};

status: {
  valid: ContrastColor;
  info: ContrastColor;
  warning: ContrastColor;
  error: ContrastColor;
};
```

#### To

```
export type ContrastColor = {
  background: string;
  text: string;
};

export type StatusColor = {
  primary: ContrastColor;
  secondary: ContrastColor;
};

status: {
  valid: StatusColor;
  info: StatusColor;
  warning: StatusColor;
  error: StatusColor;
};
```
</details>

<details>
<summary>Status color is now on its own group, no longer inside static colors</summary>

From:
```
static: {
    background: {
      background_0: ContrastColor;
      background_1: ContrastColor;
      background_2: ContrastColor;
      background_3: ContrastColor;
      background_accent_0: ContrastColor;
      background_accent_1: ContrastColor;
      background_accent_2: ContrastColor;
      background_accent_3: ContrastColor;
      background_accent_4: ContrastColor;
      background_accent_5: ContrastColor;
    };

    status: {
      valid: StatusColor;
      info: StatusColor;
      warning: StatusColor;
      error: StatusColor;
    };

    zone_selection: {
      from: ContrastColor;
      to: ContrastColor;
    };
  };
```

To:
```
 status: {
    valid: StatusColor;
    info: StatusColor;
    warning: StatusColor;
    error: StatusColor;
  };

  static: {
    background: {
      background_0: ContrastColor;
      background_1: ContrastColor;
      background_2: ContrastColor;
      background_3: ContrastColor;
      background_accent_0: ContrastColor;
      background_accent_1: ContrastColor;
      background_accent_2: ContrastColor;
      background_accent_3: ContrastColor;
      background_accent_4: ContrastColor;
      background_accent_5: ContrastColor;
    };

    zone_selection: {
      from: ContrastColor;
      to: ContrastColor;
    };
  };
```
</details>

### CODE REVIEW

**Review by rosvik — APPROVED** [review]
Looks great! Have one suggestion for refactoring the styles in MessageInfoBox, but no blocker.

File: src/components/message-info-box/MessageInfoBox.tsx [hunk 1]
```diff
@@ -51,10 +51,14 @@ export const MessageInfoBox = ({
...
   const {t} = useTranslation();
-  const iconColorProps = {fill: theme.static.status[type].text};
+  const iconColorProps = {fill: theme.status[type].secondary.text};
+  const borderColorStyle = {
+    borderColor: theme.status[type].primary.background,
+  };
   const backgroundColorStyle = {
-    backgroundColor: theme.static.status[type].background,
+    backgroundColor: theme.status[type].secondary.background,
   };
```
[the remark this answers is not in the document]
Comment (reidzeibel) [reply src/components/message-info-box/MessageInfoBox.tsx:60]: Thanks, I just followed the existing format, but I can change to that 👍🏼

[the remark this answers is not in the document]
Comment (reidzeibel) [reply src/components/message-info-box/MessageInfoBox.tsx]: Updated both MessageInfoBox and MessageSection Item, could you please do a quick check on them?

[the remark this answers is not in the document]
Comment (rosvik) [reply src/components/message-info-box/MessageInfoBox.tsx]: @reidzeibel Clean! 💯🚀

**Pull request merged**